nixos/modules/services/web-apps/languagetool.nix

```
@@ -0,0 +1,67 @@
+{ config, pkgs, lib, ... }:
+with lib;
+let
+  cfg = config.services.languageToolHttp;
+  format = pkgs.formats.toml { };
+  confDoc = "See languagetool-http-server --help for further details";
+in
+{
+  options.services.languageToolHttp = {
+    enable = mkEnableOption "the LanguageTool http server";
+    allowBrowserPluginAccess = mkEnableOption "access from the browser plugin";
+    public = mkEnableOption ''
+      access this server from anywhere, not only localhost.
+      Note that it is reccomended to run this behind
+      a HTTPS reverse proxy like apache or ngnix
+    '';
+    settings = mkOption {
+      description = ''
+        Content of the configuration file. ${confDoc}
+      '';
+      inherit (format) type;
+      default = { };
+    };
+    args = mkOption {
+      description = ''
+        Arguments to be passed when starting the server. ${confDoc}
+      '';
+      type = types.submodule {
+        freeformType = with types;attrsOf (oneOf [ int str path ]);
+        options.port = mkOption {
+          description = ''
+            The port which the LanguageTool server listen.
+          '';
+          type = types.port;
+          default = 8081;
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.languageToolHttp.args = {
+      allow-origin = mkIf cfg.allowBrowserPluginAccess (mkDefault "*");
+      config = format.generate "LanguageTool.cfg" cfg.settings;
+      public = mkIf cfg.public "";
```

Note that the documentation says the following:
```
  --allow-origin [ORIGIN] set the Access-Control-Allow-Origin header in the HTTP response,
                         used for direct (non-proxy) JavaScript-based access from browsers.
                         Example: --allow-origin "https://my-website.org"
                         Don't set a parameter for `*`, i.e. access from all websites.
```

But also note that using `mkDefault ""` will cause `Allow-Origin` to be empty. `toGNUCommandLineShell` translates `bool` values to flags (i.e. no additional parameter), so this here should work, too:

```suggestion
      allow-origin = mkDefault cfg.allowBrowserPluginAccess;
      config = format.generate "LanguageTool.cfg" cfg.settings;
      public = mkDefault cfg.public;
```